src/storage/garbage_collector.cpp

```diff
@@ -13,25 +17,43 @@
 namespace terrier::storage {
 
 std::pair<uint32_t, uint32_t> GarbageCollector::PerformGarbageCollection() {
+  // Create the UndoBuffer for this GC run
+  delta_record_compaction_buffer_ = new UndoBuffer(txn_manager_->buffer_pool_);
+  // The compaction buffer is empty
+  compaction_buffer_empty = true;
+
   ProcessDeferredActions();
   uint32_t txns_deallocated = ProcessDeallocateQueue();
   STORAGE_LOG_TRACE("GarbageCollector::PerformGarbageCollection(): txns_deallocated: {}", txns_deallocated);
   uint32_t txns_unlinked = ProcessUnlinkQueue();
   STORAGE_LOG_TRACE("GarbageCollector::PerformGarbageCollection(): txns_unlinked: {}", txns_unlinked);
-  if (txns_unlinked > 0) {
-    // Only update this field if we actually unlinked anything, otherwise we're being too conservative about when it's
-    // safe to deallocate the transactions in our queue.
-    last_unlinked_ = txn_manager_->GetTimestamp();
-  }
+  // Update the last unlinked timestamp every GC run conservatively whether or not any transaction/compacted
+  // undo record is unlinked.
```

Comment:
Style nit: maybe capitalize first character to be consistent with other comments

Reply:
The comment is a continuation of the previous sentence